torch/csrc/jit/mobile/import.cpp

```diff
@@ -285,14 +418,24 @@ mobile::Module BytecodeDeserializer::deserialize(
   // being a Tuple (int, table), and the integer stands for the bytecode version
   // number. The rest of the elements are the same as before.
   //
+  auto meta_dict = readMobileMetadata(mcu);
   auto bvals = readArchive("bytecode", mcu).toTuple()->elements();
 
   c10::optional<std::vector<IValue>> debug_info_bvals;
   if (reader_->hasRecord("mobile_debug.pkl")) {
     debug_info_bvals = readArchive("mobile_debug", mcu).toTuple()->elements();
   }
-  parseMethods(bvals, debug_info_bvals, *mcu);
-  auto meta_dict = readMobileMetadata(mcu);
+
+  std::vector<IValue> constant_values_from_jit;
+  bool need_tensor_jit = has_tensor_jit_index(bvals);
```

Comment:
Instead of doing this eagerly, would it be possible to do it lazily?

Basically create a lambda that reads the constants from jit and pass that lambda in to the deserializer. The first time the code finds a redirect entry, call the lambda and fill in the map.

I'm guessing that will also result in a smaller code footprint since all the code to check for Tensor jit index will going away.

Reply:
It's definitely one option. If we do it as the way you suggested, we still need a boolean to mark whether we find the first entry or not. Since deserializer is shared by many files, and this mark only apply to constant, I'm a bit hesitating to add extra logic, plus change the api of the deserializer only for one file.